frontend/src/pages/notebookController/StartServerModal.tsx

```diff
@@ -1,28 +1,45 @@
 import * as React from 'react';
-import {
-  ActionList,
-  ActionListItem,
-  Button,
-  Modal,
-  ModalVariant,
-  Spinner,
-} from '@patternfly/react-core';
-import NotebookControllerContext from './NotebookControllerContext';
-
+import { Alert, Button, Modal, ModalVariant, Spinner } from '@patternfly/react-core';
+import { Notebook } from '../../types';
 import './NotebookController.scss';
 
 type StartServerModalProps = {
+  isNotebookRunning: boolean;
+  notebook: Notebook | undefined;
   startShown: boolean;
   setStartModalShown: (shown: boolean) => void;
   onClose: () => void;
 };
 
+type SpawnStatus = {
+  status: 'success' | 'danger' | 'warning' | 'info' | 'default';
+  title: string;
+  reason: React.ReactNode;
+};
+
 const StartServerModal: React.FC<StartServerModalProps> = ({
+  isNotebookRunning,
+  notebook,
   startShown,
   setStartModalShown,
   onClose,
 }) => {
-  const { notebook, isNotebookRunning } = React.useContext(NotebookControllerContext);
+  const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
+
+  React.useEffect(() => {
+    if (isNotebookRunning) {
+      setSpawnStatus({
+        status: 'success',
+        title: 'Success',
+        reason: 'The notebook server is up and running. This page will update momentarily.',
+      });
+      setTimeout(() => {
+        if (notebook?.metadata.annotations?.['opendatahub.io/link']) {
+          window.location.href = notebook?.metadata.annotations?.['opendatahub.io/link'];
+        }
+      }, 6000);
+    }
+  }, [isNotebookRunning, notebook]);
```

Comment:
Any fears here with the `Notebook` reference updating? You'll be triggering a timeout frequently if that's the case.